lib/utils/datasetState.hpp

```diff
@@ -630,4 +630,111 @@ class acqDatasetIdState : public datasetState {
     }
 };
 
+
+/**
+ * @brief A dataset state that describes the gains being applied.
+ *
+ * @author Richard Shaw
+ */
+class gainState : public datasetState {
+public:
+    /**
+     * @brief Constructor
+     * @param data  The product information as serialized by
+     *              gainState::to_json().
+     */
+    gainState(const json& data) {
+        try {
+            _update_id = data["update_id"].get<std::string>();
+            _transition_interval = data["transition_interval"].get<double>();
+        } catch (std::exception& e) {
+            throw std::runtime_error(fmt::format(
+                fmt("gainState: Failure parsing json data ({:s}): {:s}"), data.dump(4), e.what()));
+        }
+    };
+
+    /**
+     * @brief Constructor
+     * @param  update_id  The string update_id labelling the applied gains.
+     * @param  transition_interval  The length of time to blend updates over.
+     */
+    gainState(std::string update_id, double transition_interval) :
+        _update_id(update_id),
+        _transition_interval(transition_interval){};
+
+    /**
+     * @brief Get the update_id
+     **/
+    const std::string& get_update_id() const {
+        return _update_id;
+    }
+
+    /**
+     * @brief Get the length of time to blend this new update with the previous one.
+     **/
+    double get_transition_interval() const {
+        return _transition_interval;
+    }
+
+private:
+    /// Serialize the data of this state in a json object
+    json data_to_json() const override {
+        json j;
+        j["update_id"] = _update_id;
+        j["transition_interval"] = _transition_interval;
+        return j;
+    }
+
+    // The label for the gains
+    std::string _update_id;
+
+    // The length of time we (in seconds) should blend gain updates over
```

Comment:
```suggestion
    // The length of time (in seconds) gain updates are blended over.
```
But is this for blending with the previous or the next update?